sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClient.java

```diff
@@ -193,32 +194,31 @@ public void uploadFromFile(String filePath, ParallelTransferOptions parallelTran
         Duration timeout) throws UncheckedIOException {
         this.uploadFromFileWithResponse(new BlobUploadFromFileOptions(filePath)
                 .setParallelTransferOptions(parallelTransferOptions).setHeaders(headers).setMetadata(metadata)
-                .setTier(tier).setRequestConditions(requestConditions),
-            timeout, null);
+                .setTier(tier).setRequestConditions(requestConditions).setTimeout(timeout),
+            null);
     }
 
     /**
      * Creates a new block blob, or updates the content of an existing block blob.
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.storage.blob.specialized.cryptography.EncryptedBlobClient.uploadFromFileWithResponse#BlobUploadFromFileOptions-Duration-Context}
+     * {@codesnippet com.azure.storage.blob.specialized.cryptography.EncryptedBlobClient.uploadFromFileWithResponse#BlobUploadFromFileOptions-Context}
      *
      * @param options {@link BlobUploadFromFileOptions}
-     * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
      * @throws UncheckedIOException If an I/O error occurs
      * @return Information about the uploaded block blob.
      */
     @Override
     public Response<BlockBlobItem> uploadFromFileWithResponse(BlobUploadFromFileOptions options,
-                                                              Duration timeout, Context context)
+                Context context)
         throws UncheckedIOException {
         Mono<Response<BlockBlobItem>> upload =
             this.encryptedBlobAsyncClient.uploadFromFileWithResponse(options)
                 .subscriberContext(FluxUtil.toReactorContext(context));
 
         try {
-            return StorageImplUtils.blockWithOptionalTimeout(upload, timeout);
+            return StorageImplUtils.blockWithOptionalTimeout(upload, options.getTimeout());
```

Comment:
I think in some APIs Duration is outside the options and in some it is in the options. We should probably figure out a consistent way to do it

Reply:
Any method that has been given an overload which accepts options has the timeout moved to the options. Any api where the timeout is still its own parameter has not yet had an options overload added. (Unless I've missed something)